Simplenote/src/main/java/com/automattic/simplenote/utils/Either.kt

```diff
@@ -0,0 +1,22 @@
+package com.automattic.simplenote.utils
+
+sealed class Either<out L, out R> {
```

Comment:
The problem with `Result` is that there is a `Result` in Kotlin's standard library and other places in the Android framework which makes the use confusing. So I am hesitant here to change it to `Result`. Do you think this is not a problem at all?

Reply:
Yes, it is definitely simpler. But for this case, the simplification doesn't work. The final operation is converging into a `CollaboratorsActionResult` type since I didn't want to use exceptions. This operation can be short-circuit If the note is in the trash or deleted. It would be more simpler if I added something like `flatMap` to handle left or right results (something like [railway programming](https://www.greenbird.com/news/railway-oriented-programming-in-kotlin)). 

I will leave like this for this PR, but I will consider changing it if I see some problems in future PRs. Also, I will keep an eye to external PRs to see if the use of `Either` was not considered because of the name of lack of knowledge on functional programming.

Reply:
👋 @danilo04 !

Once more, thanks a lot for all the clarification! 🙏 

Yes, let's then keep the `Either` for now and see how it goes...